demonstrations/tutorial_diffable_shadows.py

```diff
@@ -0,0 +1,500 @@
+
+def rmsd(x, y):
+    """root mean square difference"""
+    return np.sqrt(np.mean((x - y)**2))
+
+n_wires = 10
+
+x = np.arange(2*n_wires, dtype="float64")
+def circuit():
+    for i in range(n_wires):
+        qml.RY(x[i], i)
+    for i in range(n_wires-1):
+        qml.CNOT((i, i+1))
+    for i in range(n_wires):
+        qml.RY(x[i+n_wires], i)
+
+obs = qml.PauliX(0) @ qml.PauliZ(3) @ qml.PauliX(6) @ qml.PauliZ(7)
+
+dev_ideal = qml.device("default.qubit", wires=range(n_wires), shots=None)
+@qml.qnode(dev_ideal)
+def qnode_ideal():
+    circuit()
+    return qml.expval(obs)
+
+exact = qnode_ideal()
+
+##############################################################################
+# We now compare estimating the observable with classical shadows vs the canonical estimation.
+
+finite = []
+shadow = []
+shotss = range(100, 1000, 100)
+for shots in shotss:
+    for _ in range(10):
+        # repeating experiment 10 times to obtain averages and standard deviations
+        dev = qml.device("default.qubit", wires=range(10), shots=shots)
+
+        @qml.qnode(dev)
+        def qnode_finite():
+            circuit()
+            return qml.expval(obs)
+
+        @qml.qnode(dev)
+        def qnode_shadow():
+            circuit()
+            return qml.shadow_expval(obs)
+
+        finite.append(rmsd(qnode_finite(), exact))
+        shadow.append(rmsd(qnode_shadow(), exact))
+
+
+dq = np.array(finite).reshape(len(shotss), 10)
+dq, ddq = np.mean(dq, axis=1), np.var(dq, axis=1)
+ds = np.array(shadow).reshape(len(shotss), 10)
+ds, dds = np.mean(ds, axis=1), np.var(ds, axis=1)
+
+plt.errorbar(shotss, ds, yerr=dds, fmt="x-", label="shadow")
+plt.errorbar(shotss, dq, yerr=ddq, fmt="x-", label="direct")
+plt.xlabel("total number of shots T", fontsize=20)
+plt.ylabel("Accuracy (RMSD)", fontsize=20)
+plt.legend()
+plt.show()
+
+##############################################################################
+# Unsurprisingly, the accuracy is consistently better by directly measuring the observable since we are not discarding any measurement results.
+
+##############################################################################
+# All q-local observables
+# ~~~~~~~~~~~~~~~~~~~~~~~
+# For the case of measuring `all` q-local Pauli strings we expect both strategies to yield more or less the same results.
+# In this extreme case, no measurements are discarded in the classical shadow protocol.
+# Let us put that to test. First, we generate a list of all q-local observables for n qubits.
+
+from itertools import product, combinations
+from functools import reduce
+
+all_observables = []
+n = 5
+q = 2
+# create all combination of q entries of range(n)
+for w in combinations(range(n), q):
+    # w = [0, 1], [0, 2], .., [1, 2], [1, 3], .., [n-1, n]
+    observables = []
+
+    # Create all combinations of possible Pauli products P_i P_j P_k.... for wires w
+    for obs in product(
+        *[[qml.PauliX, qml.PauliY, qml.PauliZ] for _ in range(len(w))]
+        ):
```

Comment:
Match the indent level of the open bracket.
```suggestion
    for obs in product(
        *[[qml.PauliX, qml.PauliY, qml.PauliZ] for _ in range(len(w))]
    ):
```